main.go

```diff
@@ -247,7 +386,7 @@ func main() {
 			}
 		}
 
-		ul := &StatsDUnixgramListener{conn: uxgconn, eventHandler: eventQueue, logger: logger}
+		ul := &listener.StatsDUnixgramListener{Conn: uxgconn, EventHandler: eventQueue, Logger: logger}
```

Comment:
the unixgram listener is missing the full initialization with metrics, leading to a nil pointer dereference when the first packet is received there. could you add this please?